### ISSUE TITLE
Specifying Globals didn't work in TypeScript ESLint Parser ?

### ISSUE THREAD

**Issue by Weiyu-Chen** [eslint/eslint#10803]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 10.8.0
* **npm Version:** 6.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
typescript-eslint-parser

**Please show your full configuration:**

 

```js
module.exports = {
  root: true,
  env: {
    node: true
  },
  extends: ["plugin:vue/essential", "@vue/prettier", "@vue/typescript"],

  rules: {
 
  },
  parserOptions: {
    parser: "typescript-eslint-parser"
  },
  globals: {
    Vue: true,
  }
};

```

I use `Webpack.ProvidePlugin` to provide the variable.
```
    plugins: [
      new webpack.ProvidePlugin({
        Vue: ['vue', 'default']
      })
    ]
```
 

**What did you expect to happen?**
no error should be show?

**What actually happened? Please include the actual, raw output from ESLint.**
throw ts error : `Cannot find name 'Vue'`

**Comment by not-an-aardvark** [comment]
Hi, thanks for creating an issue.

The error `Cannot find name 'Vue'` seems like it's coming from Typescript, not ESLint. I suspect you're not configuring Typescript correctly.

Note that adding globals in your ESLint configuration doesn't affect which globals the Typescript knows about.

I also see that your config looks like this:

```
  parserOptions: {
    parser: "typescript-eslint-parser"
  },
```

The `parser` flag is independent from `parserOptions`, so it should look like this instead:

```
  parser: "typescript-eslint-parser"
  parserOptions: {
    
  },
```

(However, I'm unsure whether this is the cause of the problem you're encountering.)

**Comment by g-plane** [comment]
@not-an-aardvark I think putting `parser` field into `parserOptions` is correct if @Weiyu-Chen is using `vue-eslint-parser`.

@Weiyu-Chen Your problem may be from TypeScript not ESLint. You can try creating a file called `shims.d.ts` in your project root. Then edit this file:
```typescript
import _Vue from 'vue'

declare global {
  var Vue: typeof _Vue
}
```

**Comment by Weiyu-Chen** [comment]
@g-plane  it helps,  thanks.
